### PR TITLE
Add a test to reproduce finalizer/mustRunAfter behaviour

### PR DESCRIPTION
**Pull request by wolfs**
A reproducer for #10549

### CODE REVIEW

File: subprojects/core/src/integTest/groovy/org/gradle/api/FinalizerTaskIntegrationTest.groovy [hunk 1]
```diff
@@ -249,6 +250,56 @@ class FinalizerTaskIntegrationTest extends AbstractIntegrationSpec {
...
+                        sleep 1000
+                    }
+                    finalizedBy(finalizer)
+                }
+            }
+            
+            configure(project(':b')) {
+                task work {
+                    doLast {
+                        println "run after finalizer"
```
Comment (lptr) [inline subprojects/core/src/integTest/groovy/org/gradle/api/FinalizerTaskIntegrationTest.groovy]: This is a bit confusing for me. The imperative mode doesn't make sense to me in a printed message. Should this mean something like "this should have executed after the finalizer"?

Comment (wolfs) [reply subprojects/core/src/integTest/groovy/org/gradle/api/FinalizerTaskIntegrationTest.groovy]: I changed the message.

**Comment by wolfs** [comment]
@lptr PTAL!

**Comment by lptr** [comment]
LGTM.

**Pull request merged**